drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -208,12 +230,11 @@ static int pwm_nrf5_sw_pin_set(struct device *dev, u32_t pwm,
 static int pwm_nrf5_sw_get_cycles_per_sec(struct device *dev, u32_t pwm,
 					  u64_t *cycles)
 {
-	struct pwm_config *config;
-
-	config = (struct pwm_config *)dev->config->config_info;
-
-	/* HF timer frequency is derived from 16MHz source with a prescaler */
-	*cycles = 16000000UL / BIT(config->prescaler);
+	/* Since this function might be removed (see issue #6958), the maximum
+	 * supported frequency (16 MHz) is always returned here, and dynamically
+	 * adjusted timer prescalers are used in the pin set function.
```

Comment:
Not sure I understand this logic. Might the application want to know the actual frequency after scaling, perhaps to know the current timing range/resolution?

Reply:
We use the same approach in [pwm_nrfx](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/pwm/pwm_nrfx.c#L236). It is much more convenient for the users to have the prescaler computed dynamically (see the discussion in #5469 and the referred #6958) instead of having to set it at the build time.